### DISCUSSION TITLE
Question: STL Support?

### DISCUSSION THREAD

**Discussion by PowerBroker2** [peterdsharpe/AeroSandbox#96] [Q&A]
Is it possible to load an STL (or other CAD file) of a wing/fuselage to do aero analysis?

**Comment by peterdsharpe** [comment]
Hi @PowerBroker2 ,

Apologies for the delayed reply. Currently this is not possible. AeroSandbox comes with a few different 3D aero analysis submodules ([available here](https://github.com/peterdsharpe/AeroSandbox/tree/master/aerosandbox/aerodynamics/aero_3D)), but all are based on the `asb.Airplane` data structure.

In general, doing rapid 3D aero analysis without any "problem structure" about what's what on the geometry is really difficult (though, not impossible). The `asb.Airplane` data structure allows us to have a lot more info about this (e.g., what's a wing, what's a fuselage; where to apply different aero models; how to mesh different geometry parts in an efficient way; where to shed vortex wakes from in potential flow methods, etc.).

Supporting general 3D analysis on geometries with no problem structure is a very-long-distance goal of mine, but definitely not in the short-term roadmap due to its complexity. (As always, PRs welcome here!) Apologies for the unsatisfying answer here.
